js/src/lib/components/InferenceWidget/shared/WidgetExample.ts

```diff
@@ -12,83 +12,60 @@ export interface WidgetExampleOutputText {
 export interface WidgetExampleOutputUrl {
 	url: string;
 }
+
+export type WidgetExampleOutput =
+	| WidgetExampleOutputLabels
+	| WidgetExampleOutputAnswerScore
+	| WidgetExampleOutputText
+	| WidgetExampleOutputUrl;
 //#endregion
 
-export interface WidgetExampleBase {
+export interface WidgetExampleBase<TOutput> {
```

Comment:
```suggestion
export interface WidgetExampleBase<TOutput extends WidgetExampleOutput = WidgetExampleOutput> {
```

maybe? (haven't tried)

Reply:
or better w/o the default actually